include/deal.II/arborx/access_traits.h

```diff
@@ -0,0 +1,275 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_arborx_access_traits_h
+#define dealii_arborx_access_traits_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_ARBORX
+#  include <deal.II/base/bounding_box.h>
+
+#  include <ArborX.hpp>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace ArborXWrappers
+{
+  /**
+   *  This structure is used by ArborXWrappers::BVH to determine which points
+   *  intersect the bounding boxes used to build the ArborXWrappers::BVH.
+   */
+  struct PointIntersect
+  {
+    /**
+     * Constructor. @p dim_points is a list of points which we are interested in
+     * knowing if they intersect ArborXWrappers::BVH bounding boxes.
+     */
+    template <int dim, typename Number>
+    PointIntersect(const std::vector<dealii::Point<dim, Number>> &dim_points)
+    {
```

Comment:
Please move the implementation out-of-line. In fact, it could probably be in a `.cc` file.

Reply:
But the `.h` file is also ok, then you don't have to deal with instantiations.